contracts/superswap/RexSuperSwap.sol

```diff
@@ -0,0 +1,108 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+import { ISuperToken } from "@superfluid-finance/ethereum-contracts/contracts/interfaces/superfluid/ISuperfluid.sol";
+import "./TransferHelper.sol";
+import "./interfaces/ISwapRouter02.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "hardhat/console.sol";
+
+contract RexSuperSwap {
+  using SafeERC20 for ERC20;
+  ISwapRouter02 public immutable swapRouter;
+
+  event SuperSwapComplete(uint256 amountOut);
+  event SwapJustMade(uint256 amountIn);
+
+  constructor(ISwapRouter02 _swapRouter) {
+    swapRouter = _swapRouter;
+  }
+
+  /**
+   * @dev Swaps `amountIn` of `_from` SuperToken for at least `amountOutMin`
+   * of `_to` SuperToken through `path` with `poolFees` fees for each pair.
+   *
+   * Returns the amount of `_to` SuperToken received.
+   */
+  function swap(
+    ISuperToken _from,
+    ISuperToken _to,
+    uint256 amountIn,
+    uint256 amountOutMin,
+    address[] memory path,
+    uint24[] memory poolFees // Example: 0.3% * 10000 = 3000
+  ) external returns (uint256 amountOut) {
+    require(amountIn > 0, "Amount cannot be 0");
+    require(path.length > 1, "Incorrect path");
+    require(poolFees.length == path.length - 1, "Incorrect poolFees length");
+
+    // Step 1: Get underlying tokens and verify path
+    address fromBase = _from.getUnderlyingToken();
+    address toBase = _to.getUnderlyingToken();
```

Comment:
You're missing code here to handle native supertokens, see this code: https://github.com/Ricochet-Exchange/ricochet-protocol/blob/v2/contracts/REXTwoWayMarket.sol#L396

Reply:
Okay i've now added this check